src/cssjanus.js

```diff
@@ -116,7 +116,7 @@ function CSSJanus() {
 		nmcharPattern = '(?:[_a-z0-9-]|' + nonAsciiPattern + '|' + escapePattern + ')',
 		identPattern = '-?' + nmstartPattern + nmcharPattern + '*',
 		quantPattern = numPattern + '(?:\\s*' + unitPattern + '|' + identPattern + ')?',
-		signedQuantPattern = '((?:-?' + quantPattern + ')|(?:inherit|auto))',
+		signedQuantPattern = '((?:-?' + quantPattern + ')|(?:inherit|auto)|(?:calc\\(.*\\)))',
```

Comment:
We generally avoid `.*` as it means false matches can lead to arbitrary corruption of the stylesheet in RTL mode (which developers do not test very often), whereas the failure mode we prefer is to locally skip the rule we don't understand - as today's behaviour.

This way, there can generally not be production-blocking bugs in CSSJanus, and there's always a way for developers to get work done by leaving something unflipped temporarily or (if critical) to write an extra RTL-specific CSS rule by hand.

Consider these two examples:

```css
.foo { padding: 1px calc( ( 1 + 1 ) * 1px ) 3px 4px; }
.bar { padding: 1px calc( 1 * 1px ) 3px 4px; content: attr(data-title); }
```

For each declaration within a rule individually, acceptable outputs are:

* match and flip nothing.
* flip in a syntactically correct way.

With `.*` we either break the first one (stop at first closing paren), or break the second (eat through other rules until the last closing paren).

I suggest as first version we support only simple expressions and thus specify what kind of syntax can be used inside `calc()` and match those, possibly using some of the other patterns, or through an exclusion pattern like `[^;)}]*` which would accept anything that isn't one of `;)}`.

Reply:
I took notes of that,
now it have:
- support multi braces
- simpler regex
- exact calc group matching